src/odemis/driver/piezomotor.py

```diff
@@ -0,0 +1,733 @@
+                encoder_cnts = round(val / ENCODER_STEPSIZE)
+                # There are two possibilities: move relative to current position (XC) and move relative to
+                # target position (XR). We are moving relative to the current position (might be more intuitive
+                # if something goes wrong and we're stuck in the wrong position).
+                self._sendCommand(b'X%dC%d,%d' % (self._axis_map[axis], encoder_cnts, speed_usteps))
+                targets[axis] = self.position.value[axis] / ENCODER_STEPSIZE + encoder_cnts
+            else:
+                wfm_steps = int(val / self._wfm_stepsizes[axis])  # number of waveform steps
+                usteps = int((val % self._wfm_stepsizes[axis]) * USTEPS_PER_WFM)   # number of µsteps
+                self._sendCommand(b'X%dJ%d,%d,%d' % (self._axis_map[axis], wfm_steps, usteps, speed_usteps))
+                targets[axis] = None
+        self._waitEndMotion(targets)
+        self._updatePosition()
+
+    def _waitEndMotion(self, targets):
+        """
+        Wait until move is done
+        :arg targets (dict: str --> int): target (for closed-loop), None for open loop
+        """
+        # Expected time for move
+        move_length = sum([abs(self.position[ax] - targets[ax]) for ax in targets.keys()])
+        dur = move_length * self._speed
+        max_dur = max(dur * 2, 0.1)  # wait at least 0.1 s
+        logging.debug("Expecting a move of %g s, will wait up to %g s", dur, max_dur)
+
+        for ax, target in targets.items():
+            moving = True
+            end_time = time.time() + 5  # 5 s until timeout
+            while moving:
+                if time.time() > end_time:
+                    raise IOError("Timeout while waiting for end of motion on axis %s" % ax)
+                if self._closed_loop[ax]:
+                    if target is None:
+                        raise ValueError("No target provided for closed-loop move on axis %s." % ax)
+                    moving = self.isMovingClosedLoop(ax, target)
+                else:
+                    moving = self.isMovingOpenLoop(ax)
+                self._check_hw_error()
+                time.sleep(0.05)
+
+    def _check_hw_error(self):
+        for ax, axnum in self._axis_map.items():
+            status = self.getStatus(axnum)
+            if status[0] == 8:
+                raise PMDError(1, "Communication Error (wrong baudrate, data collision, or buffer overflow)")
+            elif status[0] == 4:
+                raise PMDError(2, "Encoder error(serial communication or reported error from serial encoder)")
+            elif status[0] == 2:
+                raise PMDError(3, "Supply voltage or motor fault was detected.")
+            elif status[1] == 1:
+                raise PMDError(4, "Command timeout occurred or a syntax error was detected when response was not allowed.")
+            elif status[1] == 8:
+                raise PMDError(5, "Power-on/reset has occurred.")
```

Comment:
You need to use bit comparison, not equality, because there might be several errors at the same time. => Replace `==` by `&`.